src/main/java/org/opensearch/ad/indices/AnomalyDetectionIndices.java

```diff
@@ -1072,68 +1082,79 @@ public int getSchemaVersion(ADIndex index) {
+                        IndexMetadata.SETTING_AUTO_EXPAND_REPLICAS
+                    )
+            )
+            .build();
+
+        CompletableFuture<GetIndicesSettingsResponse> getIndicesSettingsResponse = sdkJavaAsyncClient
+            ._transport()
+            .performRequestAsync(getIndicesSettingsRequest, GetIndicesSettingsRequest._ENDPOINT, TransportOptions.builder().build());
+
+        GetIndicesSettingsResponse settingResponse;
+        try {
+            settingResponse = getIndicesSettingsResponse.orTimeout(10L, TimeUnit.SECONDS).get();
+            // auto expand setting is a range string like "1-all"
+            org.opensearch.client.opensearch.indices.IndexState indexState = settingResponse.get(ADIndex.JOB.getIndexName());
+            String autoExpandReplica = indexState.settings().autoExpandReplicas();
+            // if the auto expand setting is already there, return immediately
+            if (autoExpandReplica != null) {
+                jobIndexState.settingUpToDate = true;
+                logger.info(new ParameterizedMessage("Mark [{}]'s mapping up-to-date", ADIndex.JOB.getIndexName()));
+                listener.onResponse(null);
+                return;
+            }
+            Integer primaryShardsNumber = Integer.valueOf(indexState.settings().numberOfShards());
+            Integer replicaNumber = Integer.valueOf(indexState.settings().numberOfReplicas());
+            if (primaryShardsNumber == null || replicaNumber == null) {
+                logger
+                    .error(
+                        new ParameterizedMessage(
+                            "Fail to find AD job index's primary or replica shard number: primary [{}], replica [{}]",
+                            primaryShardsNumber,
+                            replicaNumber
+                        )
+                    );
+                // don't throw exception as we don't know how to handle it and retry next time
+                listener.onResponse(null);
+                return;
+            }
+            // at least minJobIndexReplicas
+            // at most maxJobIndexReplicas / primaryShardsNumber replicas.
+            // For example, if we have 2 primary shards, since the max number of shards are maxJobIndexReplicas (20),
+            // we will use 20 / 2 = 10 replicas as the upper bound of replica.
+            int maxExpectedReplicas = Math.max(maxJobIndexReplicas / primaryShardsNumber, minJobIndexReplicas);
+            Settings updatedSettings = Settings
+                .builder()
+                .put(IndexMetadata.SETTING_AUTO_EXPAND_REPLICAS, minJobIndexReplicas + "-" + maxExpectedReplicas)
+                .build();
+            final UpdateSettingsRequest updateSettingsRequest = new UpdateSettingsRequest(ADIndex.JOB.getIndexName())
+                .settings(updatedSettings);
+            client.indices().putSettings(updateSettingsRequest, ActionListener.wrap(response -> {
```

Comment:
We are making an admin call here. We do have `admin` in client. Let's make sure we request carefully here and for the rest of the migration.
```suggestion
            client.admin().indices().putSettings(updateSettingsRequest, ActionListener.wrap(response -> {
```

Reply:
Ill just set adminClient to sdkRestClient.admin()